crates/red_knot/src/main.rs

```diff
@@ -48,225 +44,305 @@ fn main() -> anyhow::Result<()> {
         return Err(anyhow::anyhow!("Invalid arguments"));
     }
 
-    let files = files::Files::default();
     let workspace_folder = entry_point.parent().unwrap();
-    let mut workspace = Workspace::new(workspace_folder.to_path_buf());
+    let workspace = Workspace::new(workspace_folder.to_path_buf());
```

Comment:
It's not clear to me that `Workspace` shouldn't just be rolled directly into `Program` instead of being a separate struct. But maybe there will be benefits in having `Workspace` separate.

Reply:
Yeah, `workspace` might just be rolled into `Program`. Although it's somewhat a different entity in that it doesn't represent the program state, but the project state. For example, we could discover all files in a workspace (which is different from all files in a Program because the workspace does not include third party dependencies), what configurations exist etc independent of the program.